main.js

```diff
@@ -0,0 +1,92 @@
+//this function creates the collapsible navbar
+$( document ).ready(function(){
+  $(".button-collapse").sideNav();
+});
+
+
+//this first clickk event does everything when you click a card to build your order
+$('.card').on('click', 'a', function(event) {
+  console.log('clicking');
+
+  addItemToOrder(getOrderData(event, '.card-title'), getOrderData(event, '.card-price'));
+  // updateSubtotal(event);
+  updateTaxAndTotal(updateSubtotal(event));
+});
+
+//this function adds ordered items to the order list on the right side of the page
+function addItemToOrder(name, price) {
+  const $tr = $('<tr>');
+  const $item = $('<td>').text(name)
+  const $price = $('<td>').text(price).addClass('right order-price');
+  const $orderTable = $('.order-table').children('tbody');
+  $tr.append($item);
+  $tr.append($price);
+  $orderTable.append($tr);
+}
+
+
+//this function grabs the desired data from the clicked card
+function getOrderData(event, selector) {
+  const orderData = $(event.target).parents('.card').children().children(selector).text();
+  return orderData;
+}
+
+//this function finds the prices in the order form and updates the subtotal
+function updateSubtotal(event) {
+  const orderPrices = $('.order-price');
+  let pricesArray = [];
+  for (let i = 0; i < orderPrices.length; i++) {
+    pricesArray.push(orderPrices.eq(i).text());
+  }
+  const subtotalValue = sumPrices(pricesArray);
+  console.log(subtotalValue);
+  const subtotalLocation = $('.subtotal-amount');
+  subtotalLocation.text(subtotalValue);
+  return subtotalValue;
+}
```

Comment:
you are changing the subtotal text and returning the subtotalValue. that is a side effect, which is when you do something that affects the program besides returning a value. Side effects can cause hard to track down problems. 

In this case, you could have returned the subtotalValue, and updated the DOM in the function that invoked `updateSubtotal()`